src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -1 +1,166 @@
+        state::{
+            contract_storage_state::ContractStorageState,
+            state_api::{State, StateReader},
+        },
+    },
+    core::{
+        errors::syscall_handler_errors::SyscallHandlerError,
+        syscalls::{
+            syscall_request::SyscallRequest,
+            syscall_response::{CallContractResponse, ResponseBody},
+        },
+    },
+    definitions::general_config::StarknetGeneralConfig,
+    services::api::contract_class::EntryPointType,
+    utils::{get_felt_range, Address},
+};
+
+use super::{
+    syscall_handler::SyscallHandler, syscall_info::get_syscall_size_from_name,
+    syscall_response::SyscallResponse,
+};
+
+pub struct BusinessLogicSyscallHandler<'a, T: Default + State + StateReader> {
+    pub(crate) tx_execution_context: TransactionExecutionContext,
+    /// A list of dynamically allocated segments that are expected to be read-only.
+    pub(crate) read_only_segments: Vec<(Relocatable, MaybeRelocatable)>,
+    pub(crate) resources_manager: ExecutionResourcesManager,
+    pub(crate) expected_syscall_ptr: Relocatable,
+    pub(crate) caller_address: Address,
+    pub(crate) general_config: StarknetGeneralConfig,
+    pub(crate) starknet_storage_state: ContractStorageState<'a, T>,
+    pub(crate) internal_calls: Vec<CallInfo>,
+}
+
+impl<'a, T: Default + State + StateReader> BusinessLogicSyscallHandler<'a, T> {
+    pub fn new(
+        tx_execution_context: TransactionExecutionContext,
+        state: &'a mut T,
+        contract_address: Address,
+        resources_manager: ExecutionResourcesManager,
+        expected_syscall_ptr: Relocatable,
+        caller_address: Address,
+        general_config: StarknetGeneralConfig,
+    ) -> Self {
+        let read_only_segments = vec![];
+        let starknet_storage_state = ContractStorageState::new(state, contract_address);
+        let internal_calls = vec![];
+        Self {
+            tx_execution_context,
+            read_only_segments,
+            resources_manager,
+            expected_syscall_ptr,
+            caller_address,
+            general_config,
+            starknet_storage_state,
+            internal_calls,
+        }
+    }
+
+    fn increment_syscall_count(&mut self, syscall_name: &str) {
+        self.resources_manager
+            .increment_syscall_counter(syscall_name, 1);
+    }
+
+    fn allocate_segment(
+        &mut self,
+        vm: &mut VirtualMachine,
+        data: Vec<MaybeRelocatable>,
+    ) -> Result<Relocatable, SyscallHandlerError> {
+        let segment_start = vm.add_memory_segment();
+        let segment_end = vm.write_arg(segment_start, &data)?;
+        let sub = segment_end.sub(&segment_start.to_owned().into())?;
+        let segment = (segment_start.to_owned(), sub);
+        self.read_only_segments.push(segment);
+
+        Ok(segment_start)
+    }
+}
+
+impl<'a, T: Default + State + StateReader> SyscallHandler for BusinessLogicSyscallHandler<'a, T> {
+    #[allow(irrefutable_let_patterns)]
```

Comment:
This is needed for now, when we get more types of SyscallRequest we can remove the annotation